### PR TITLE
segment controller validation 

### PR DESCRIPTION
**Pull request by Yagnik56**
This PR contains segment validation related changes.

### CODE REVIEW

File: backend/packages/Upgrade/src/api/controllers/validators/SegmentInputValidator.ts [hunk 1]
```diff
@@ -25,7 +38,9 @@ export class SegmentInputValidator {
   public userIds: string[];
```
Comment (RidhamShah) [inline backend/packages/Upgrade/src/api/controllers/validators/SegmentInputValidator.ts:38]: Array validations on `userIds` is remaining.

Comment (Yagnik56) [reply backend/packages/Upgrade/src/api/controllers/validators/SegmentInputValidator.ts:38]: it was already there

File: backend/packages/Upgrade/src/api/controllers/validators/SegmentInputValidator.ts [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
+  @IsString()
   public id: string;
 
   @IsNotEmpty()
   @IsString()
   public name: string;
 
   @IsString()
+  @IsOptional()
   public description: string;
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/controllers/validators/SegmentInputValidator.ts:26]: here -> description?

**Comment by VivekFitkariwala** [comment]
Looks good

File: backend/packages/Upgrade/src/api/controllers/validators/SegmentInputValidator.ts (hunk 2 above)
Comment (RidhamShah) [reply backend/packages/Upgrade/src/api/controllers/validators/SegmentInputValidator.ts]: Code updated

**Pull request merged**